drivers/display/display_st730x.c

```diff
@@ -0,0 +1,561 @@
+	if (err < 0) {
+		return err;
+	}
+
+	err = mipi_dbi_command_write(config->mipi_dev, &config->dbi_config,
+				     ST730X_MADCTL, &config->remap_value, 1);
+	if (err < 0) {
+		return err;
+	}
+
+	tmp[0] = ST730X_DTFORM_3W_24B;
+	err = mipi_dbi_command_write(config->mipi_dev, &config->dbi_config,
+				     ST730X_DTFORM, tmp, 1);
+	if (err < 0) {
+		return err;
+	}
+
+	tmp[0] = ST730X_GAMAMS_MONO;
+	err = mipi_dbi_command_write(config->mipi_dev, &config->dbi_config,
+				     ST730X_GAMAMS, tmp, 1);
+	if (err < 0) {
+		return err;
+	}
+
+	err = mipi_dbi_command_write(config->mipi_dev, &config->dbi_config,
+				     ST730X_PNLSET, &config->panel_settings, 1);
+	if (err < 0) {
+		return err;
+	}
+
+	tmp[0] = ST730X_TEARING_OUT_VBLANK;
+	err = mipi_dbi_command_write(config->mipi_dev, &config->dbi_config,
+				     ST730X_TEARING_OUT, tmp, 1);
+	if (err < 0) {
+		return err;
+	}
+
+	tmp[0] = ST730X_AUTOPWRDOWN_ON;
+	err = mipi_dbi_command_write(config->mipi_dev, &config->dbi_config,
+				     ST730X_AUTOPWRDOWN, tmp, 1);
+	if (err < 0) {
+		return err;
+	}
+
+	tmp[0] = (config->start_line & 0x100) >> 8;
+	tmp[1] = config->start_line & 0xFF;
+	err = mipi_dbi_command_write(config->mipi_dev, &config->dbi_config,
+				     ST730X_FIRSTGATE, tmp, 2);
+	if (err < 0) {
+		return err;
+	}
+
+	return mipi_dbi_command_write(config->mipi_dev, &config->dbi_config,
+				     ST730X_POWER_MODE, NULL, 0);
+}
+
+/* Convert what the conversion buffer can hold to the st730x format */
+static int st730x_convert(const struct device *dev, const uint8_t *buf, uint32_t offset,
+			       const struct display_buffer_descriptor *desc)
+{
+	const struct st730x_config *config = dev->config;
+	uint32_t vertical_offset = desc->width / 8;
+	uint32_t i = 0;
+	uint32_t ipos, ipos_zeroed;
+	uint32_t max_lines = (config->conversion_buf_size / (desc->width / 8)) & ~0x1;
+
+	if (max_lines < 2) {
+		LOG_ERR("Buffer too small, cannot convert");
+		return -EINVAL;
+	}
+
+	for (; (offset + i) < desc->height && i < max_lines; i += 2) {
+		ipos = (offset + i) * vertical_offset;
+		ipos_zeroed = i * vertical_offset / 2;
+		for (uint32_t j = 0; j < desc->width / 8 ; j++) {
+			config->conversion_buf[(ipos_zeroed + j) * 2 + 1] =
+				(buf[ipos + j + vertical_offset] & 0x80) >> 7
+				| (buf[ipos + j] & 0x80) >> 6
+				| (buf[ipos + j + vertical_offset] & 0x40) >> 4
+				| (buf[ipos + j] & 0x40) >> 3
+				| (buf[ipos + j + vertical_offset] & 0x20) >> 1
+				| (buf[ipos + j] & 0x20)
+				| (buf[ipos + j + vertical_offset] & 0x10) << 2
+				| (buf[ipos + j] & 0x10) << 3;
+			config->conversion_buf[(ipos_zeroed + j) * 2+0] =
+				(buf[ipos + j + vertical_offset] & 0x8) >> 3
+				| (buf[ipos + j] & 0x8) >> 2
+				| (buf[ipos + j + vertical_offset] & 0x4)
+				| (buf[ipos + j] & 0x4) << 1
+				| (buf[ipos + j + vertical_offset] & 0x2) << 3
+				| (buf[ipos + j] & 0x2) << 4
+				| (buf[ipos + j + vertical_offset] & 0x1) << 6
+				| (buf[ipos + j] & 0x1) << 7;
+		}
+	}
```

Comment:
Thanks for hiding the evilness of this pixel format so that users never have to worry about it.